feature_list/feature_list.cpp

```diff
@@ -0,0 +1,267 @@
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "geometry/mercator.hpp"
+#include "geometry/point2d.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/feature.hpp"
+#include "indexer/feature_processor.hpp"
+#include "indexer/map_object.hpp"
+#include "indexer/map_style_reader.hpp"
+
+#include "platform/local_country_file_utils.hpp"
+#include "platform/platform.hpp"
+
+#include "search/engine.hpp"
+#include "search/reverse_geocoder.hpp"
+#include "search/search_quality/helpers.hpp"
+
+#include "std/iostream.hpp"
+
+#include "storage/index.hpp"
+#include "storage/storage.hpp"
+
+class ClosestPoint
+{
+  m2::PointD const & m_center;
+  m2::PointD m_best;
+  double m_distance = -1;
+
+public:
+  ClosestPoint(m2::PointD const & center) : m_center(center) {}
+  m2::PointD GetBest() const { return m_best; }
+  void operator()(m2::PointD const & point)
+  {
+    double distance = m_center.SquareLength(point);
+    if (m_distance < 0 || distance < m_distance)
+    {
+      m_distance = distance;
+      m_best = point;
+    }
+  }
+};
+
+m2::PointD FindCenter(FeatureType const & f)
+{
+  ClosestPoint closest(f.GetLimitRect(FeatureType::BEST_GEOMETRY).Center());
+  if (f.GetFeatureType() == feature::GEOM_AREA)
+  {
+    f.ForEachTriangle([&closest](m2::PointD const & p1, m2::PointD const & p2,
+                                 m2::PointD const & p3) { closest((p1 + p2 + p3) / 3); },
+                      FeatureType::BEST_GEOMETRY);
+  }
+  else
+  {
+    f.ForEachPoint(closest, FeatureType::BEST_GEOMETRY);
+  }
+  return closest.GetBest();
+}
+
+string GetReadableType(FeatureType const & f)
+{
+  uint32_t result = 0;
+  f.ForEachType([&result](uint32_t type)
+  {
+    string fullName = classif().GetFullObjectName(type);
+    auto pos = fullName.find("|");
+    if (pos != string::npos)
+      fullName = fullName.substr(0, pos);
+    if (fullName == "amenity" || fullName == "shop" || fullName == "tourism" ||
+        fullName == "leisure" || fullName == "craft" || fullName == "place" ||
+        fullName == "man_made" || fullName == "emergency" || fullName == "historic" ||
+        fullName == "railway" || fullName == "highway" || fullName == "aeroway")
+      result = type;
```

Comment:
Curly braces, please. Or, even better, create a global vector of these strings and use find.